OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -3049,22 +3063,37 @@ public static DreamValue NativeProc_viewers(NativeProc.Bundle bundle, DreamObjec
                 }
             }
 
-            center ??= usr;
+            center ??= usr as DreamObjectAtom;
 
             DreamList view = bundle.ObjectTree.CreateList();
             if (center == null)
                 return new(view);
 
-            int centerX = center.GetVariable("x").MustGetValueAsInteger();
-            int centerY = center.GetVariable("y").MustGetValueAsInteger();
+            var centerPos = bundle.AtomManager.GetAtomPosition(center);
             if (!depthValue.TryGetValueAsInteger(out var depth))
                 depth = 5; //TODO: Default to world.view
 
             foreach (var atom in bundle.AtomManager.EnumerateAtoms(bundle.ObjectTree.Mob)) {
                 var mob = (DreamObjectMob)atom;
 
-                if (Math.Abs(centerX - mob.X) <= depth && Math.Abs(centerY - mob.Y) <= depth) {
-                    view.AddValue(new DreamValue(mob));
+                if (Math.Abs(centerPos.X - mob.X) <= depth && Math.Abs(centerPos.Y - mob.Y) <= depth && centerPos.Z == mob.Z) {
+                    (_, ViewRange range) = DreamProcNativeHelpers.ResolveViewArguments(bundle.DreamManager, mob, bundle.Arguments);
+                    var eyePos = bundle.AtomManager.GetAtomPosition(mob);
+                    var viewData = DreamProcNativeHelpers.CollectViewData(bundle.AtomManager, bundle.MapManager, eyePos, range);
+
+                    ViewAlgorithm.CalculateVisibility(viewData);
+
+                    foreach (var tile in DreamProcNativeHelpers.MakeViewSpiral(viewData, true)) {
```

Comment:
According to this, `viewers()` does not go in an outward spiral like `view()` does. It instead goes by column.
![image](https://github.com/OpenDreamProject/OpenDream/assets/30789242/4d695e26-333d-4bdc-af99-dfca342974de)

Reply:
lol

Reply:
lmao